<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BallsWindow</class>
 <widget class="QMainWindow" name="BallsWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>672</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BALLS</string>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonIconOnly</enum>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <property name="dockNestingEnabled">
   <bool>false</bool>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks</set>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="minimumSize">
    <size>
     <width>400</width>
     <height>238</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>65535</width>
     <height>65535</height>
    </size>
   </property>
   <layout class="QGridLayout" name="windowLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <property name="spacing">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="balls::BallsCanvas" name="canvas">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1000</width>
     <height>31</height>
    </rect>
   </property>
   <property name="nativeMenuBar">
    <bool>true</bool>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuExamples">
     <property name="title">
      <string>&amp;Examples</string>
     </property>
     <property name="icon">
      <iconset resource="resources.qrc">
       <normaloff>:/icons/applications-accessories.png</normaloff>:/icons/applications-accessories.png</iconset>
     </property>
     <addaction name="actionDefault"/>
     <addaction name="actionPhong_Lighting"/>
    </widget>
    <addaction name="actionNew_Project"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave_File"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_Project"/>
    <addaction name="actionSave_Project_As"/>
    <addaction name="separator"/>
    <addaction name="menuExamples"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuProject">
    <property name="title">
     <string>Pro&amp;ject</string>
    </property>
    <addaction name="actionCompile"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionAbout_BALLS"/>
    <addaction name="actionAbout_Qt"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="actionZoom_In"/>
    <addaction name="actionZoom_Out"/>
    <addaction name="actionReset_Zoom"/>
    <addaction name="separator"/>
    <addaction name="actionReset_Camera"/>
    <addaction name="separator"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuProject"/>
   <addaction name="menuView"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QDockWidget" name="editorDock">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Edito&amp;r</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="editorDockContents">
    <property name="sizePolicy">
     <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <layout class="QVBoxLayout" name="editorDockLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QTabWidget" name="tabs">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="tabPosition">
        <enum>QTabWidget::North</enum>
       </property>
       <property name="tabShape">
        <enum>QTabWidget::Rounded</enum>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <property name="elideMode">
        <enum>Qt::ElideRight</enum>
       </property>
       <property name="usesScrollButtons">
        <bool>false</bool>
       </property>
       <property name="documentMode">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="vertexTab">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <attribute name="title">
         <string>Vertex</string>
        </attribute>
        <layout class="QVBoxLayout" name="vertexLayout">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>6</number>
         </property>
         <property name="topMargin">
          <number>6</number>
         </property>
         <property name="rightMargin">
          <number>6</number>
         </property>
         <property name="bottomMargin">
          <number>6</number>
         </property>
         <item>
          <widget class="QsciScintilla" name="vertexEditor">
           <property name="font">
            <font>
             <family>DejaVu Sans Mono</family>
            </font>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="fragmentTab">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <attribute name="title">
         <string>Fragment</string>
        </attribute>
        <layout class="QVBoxLayout" name="fragmentLayout">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>6</number>
         </property>
         <property name="topMargin">
          <number>6</number>
         </property>
         <property name="rightMargin">
          <number>6</number>
         </property>
         <property name="bottomMargin">
          <number>6</number>
         </property>
         <item>
          <widget class="QsciScintilla" name="fragmentEditor">
           <property name="font">
            <font>
             <family>DejaVu Sans Mono</family>
            </font>
           </property>
           <property name="toolTip">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="geometryTab">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <attribute name="title">
         <string>Geometry</string>
        </attribute>
        <layout class="QVBoxLayout" name="geometryLayout">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>6</number>
         </property>
         <property name="topMargin">
          <number>6</number>
         </property>
         <property name="rightMargin">
          <number>6</number>
         </property>
         <property name="bottomMargin">
          <number>6</number>
         </property>
         <item>
          <widget class="QsciScintilla" name="geometryEditor">
           <property name="font">
            <font>
             <family>DejaVu Sans Mono</family>
            </font>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="uniformDock">
   <property name="windowTitle">
    <string>&amp;Uniforms</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="uniformDockContents">
    <layout class="QGridLayout" name="gridLayout_2">
     <property name="topMargin">
      <number>0</number>
     </property>
     <item row="0" column="0">
      <widget class="QScrollArea" name="uniformScroll">
       <property name="frameShape">
        <enum>QFrame::Panel</enum>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="uniformScrollContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>256</width>
          <height>191</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="uniformScrollCotentsLayout">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QPropertyEditorWidget" name="uniforms"/>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="textureDock">
   <property name="windowTitle">
    <string>Te&amp;xtures</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="balls::TextureManager" name="textureManager"/>
  </widget>
  <widget class="QDockWidget" name="meshDock">
   <property name="windowTitle">
    <string>&amp;Meshes</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="balls::MeshManagerWidget" name="meshManager">
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>18</height>
     </size>
    </property>
   </widget>
  </widget>
  <action name="actionNew_Project">
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>&amp;Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save &amp;File As...</string>
   </property>
  </action>
  <action name="actionSave_File">
   <property name="text">
    <string>&amp;Save File</string>
   </property>
  </action>
  <action name="actionSave_Project">
   <property name="text">
    <string>Save &amp;Project</string>
   </property>
  </action>
  <action name="actionSave_Project_As">
   <property name="text">
    <string>Sa&amp;ve Project As...</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/system-log-out.png</normaloff>
     <normalon>:/icons/system-log-out.png</normalon>:/icons/system-log-out.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="menuRole">
    <enum>QAction::QuitRole</enum>
   </property>
  </action>
  <action name="actionZoom_In">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/list-add.png</normaloff>:/icons/list-add.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Zoom Editor In</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+=</string>
   </property>
  </action>
  <action name="actionZoom_Out">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/list-remove.png</normaloff>:/icons/list-remove.png</iconset>
   </property>
   <property name="text">
    <string>Zoom Editor &amp;Out</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+-</string>
   </property>
  </action>
  <action name="actionReset_Zoom">
   <property name="text">
    <string>&amp;Reset Editor Zoom</string>
   </property>
  </action>
  <action name="actionCompile">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/view-refresh.png</normaloff>:/icons/view-refresh.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Compile</string>
   </property>
   <property name="shortcut">
    <string>F9</string>
   </property>
  </action>
  <action name="actionAbout_BALLS">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/help-browser.png</normaloff>:/icons/help-browser.png</iconset>
   </property>
   <property name="text">
    <string>&amp;About BALLS...</string>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutRole</enum>
   </property>
  </action>
  <action name="actionReset_Camera">
   <property name="text">
    <string>Reset &amp;Camera</string>
   </property>
  </action>
  <action name="actionOpenGL_Info">
   <property name="text">
    <string>OpenGL Info...</string>
   </property>
  </action>
  <action name="actionList_of_Uniforms">
   <property name="text">
    <string>Uniform List</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/help-browser.png</normaloff>:/icons/help-browser.png</iconset>
   </property>
   <property name="text">
    <string>About &amp;Qt...</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutQtRole</enum>
   </property>
  </action>
  <action name="actionDefault">
   <property name="text">
    <string>&amp;Default</string>
   </property>
   <property name="example" stdset="0">
    <string notr="true">default.balls</string>
   </property>
  </action>
  <action name="actionPhong_Lighting">
   <property name="text">
    <string>&amp;Phong-Blinn</string>
   </property>
   <property name="example" stdset="0">
    <string notr="true">phong.balls</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QsciScintilla</class>
   <extends>QFrame</extends>
   <header>Qsci/qsciscintilla.h</header>
  </customwidget>
  <customwidget>
   <class>QPropertyEditorWidget</class>
   <extends>QTreeView</extends>
   <header>QPropertyEditor/QPropertyEditorWidget.h</header>
   <slots>
    <slot>addObject(QObject*)</slot>
    <slot>setObject(QObject*)</slot>
    <slot>updateObject(QObject*)</slot>
    <slot>removeObject(QObject*)</slot>
    <slot>setObject(QListWidgetItem*)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>balls::BallsCanvas</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">ui/BallsCanvas.hpp</header>
   <slots>
    <signal>geometryShadersSupported(bool)</signal>
    <signal>finishedInitializing()</signal>
    <signal>graphicsWarning(QString,QString)</signal>
    <signal>fatalGraphicsError(QString,QString,int)</signal>
    <signal>updateUniform(QObject*)</signal>
    <signal>addUniform(QObject*)</signal>
    <signal>removeUniform(QObject*)</signal>
    <slot>setMesh(balls::Mesh)</slot>
    <slot>resetCamera()</slot>
    <slot>setOption(bool)</slot>
    <slot>setOption(int)</slot>
    <slot>setUniform(QVariant)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>balls::TextureManager</class>
   <extends>QWidget</extends>
   <header location="global">ui/docks/TextureManager.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>balls::MeshManagerWidget</class>
   <extends>QWidget</extends>
   <header location="global">ui/docks/MeshManagerWidget.hpp</header>
   <container>1</container>
   <slots>
    <signal>meshCreated(balls::Mesh)</signal>
    <signal>meshSelected(balls::Mesh)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionCompile</sender>
   <signal>triggered()</signal>
   <receiver>BallsWindow</receiver>
   <slot>forceShaderUpdate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>265</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>canvas</sender>
   <signal>graphicsWarning(QString,QString)</signal>
   <receiver>BallsWindow</receiver>
   <slot>reportWarning(QString,QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>70</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>canvas</sender>
   <signal>fatalGraphicsError(QString,QString,int)</signal>
   <receiver>BallsWindow</receiver>
   <slot>reportFatalError(QString,QString,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>70</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionReset_Camera</sender>
   <signal>triggered()</signal>
   <receiver>canvas</receiver>
   <slot>resetCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>112</x>
     <y>70</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoom_Out</sender>
   <signal>triggered()</signal>
   <receiver>fragmentEditor</receiver>
   <slot>zoomOut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>146</x>
     <y>209</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoom_In</sender>
   <signal>triggered()</signal>
   <receiver>fragmentEditor</receiver>
   <slot>zoomIn()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>146</x>
     <y>209</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoom_Out</sender>
   <signal>triggered()</signal>
   <receiver>geometryEditor</receiver>
   <slot>zoomOut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>146</x>
     <y>209</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout_Qt</sender>
   <signal>triggered()</signal>
   <receiver>BallsWindow</receiver>
   <slot>showAboutQt()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>414</x>
     <y>311</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDefault</sender>
   <signal>triggered()</signal>
   <receiver>BallsWindow</receiver>
   <slot>loadExample()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>319</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPhong_Lighting</sender>
   <signal>triggered()</signal>
   <receiver>BallsWindow</receiver>
   <slot>loadExample()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>319</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoom_In</sender>
   <signal>triggered()</signal>
   <receiver>vertexEditor</receiver>
   <slot>zoomIn()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>146</x>
     <y>209</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoom_Out</sender>
   <signal>triggered()</signal>
   <receiver>vertexEditor</receiver>
   <slot>zoomOut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>146</x>
     <y>209</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_Project</sender>
   <signal>triggered()</signal>
   <receiver>BallsWindow</receiver>
   <slot>saveProject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>319</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>BallsWindow</receiver>
   <slot>loadProject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>319</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setMesh(int)</slot>
  <slot>initializeMeshGenerators()</slot>
  <slot>forceShaderUpdate()</slot>
  <slot>reportFatalError(QString,QString,int)</slot>
  <slot>reportWarning(QString,QString)</slot>
  <slot>showAboutQt()</slot>
  <slot>loadExample()</slot>
  <slot>saveProject()</slot>
  <slot>loadProject()</slot>
 </slots>
</ui>
